### PR TITLE
cmdliner.0.9.7: backport to 3.12.1.

### PR DESCRIPTION
**Pull request by dbuenzli**
Fixes dbuenzli/cmdliner#22.

### CODE REVIEW

**Comment by dbuenzli** [comment]
Failures are only on opam `1.1.0` (rewrite errors ?) and `1.2.1` (?). 

At least 3.12.1 on opam 1.2.0 succeded, which was the goal.

File: packages/cmdliner/cmdliner.0.9.7/files/backport_pre_4_00_0.patch [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
++    let a = { a with absent = Val (lazy ""); o_all = true;
+                      o_kind = match vopt with
+                      | None -> Opt | Some dv -> Opt_vopt (str_of_pp print dv) }
+     in
+@@ -949,7 +949,7 @@
+   let pos ?(rev = false) k (parse, print) v a =
+     if is_opt a then invalid_arg err_not_pos else
+     let a = { a with p_kind = Nth (rev, k);
+-                     absent = Val (Lazy.from_val (str_of_pp print v)) }
++                     absent = Val (lazy (str_of_pp print v)) }
```
Comment (dbuenzli) [inline packages/cmdliner/cmdliner.0.9.7/files/backport_pre_4_00_0.patch]: Actually that's the way it should be. Corrected on master.

**Comment by samoht** [comment]
```
[default] /home/travis/build/ocaml/opam-repository synchronized
[ERROR] File
        "/home/travis/.opam/repo/default/packages/cmdliner/cmdliner.0.9.7/opam",
        line 11, character 63-64: parse error.
```
?

File: packages/cmdliner/cmdliner.0.9.7/opam [hunk 2]
```diff
@@ -7,7 +7,8 @@ dev-repo: "http://erratique.ch/repos/cmdliner.git"
 bug-reports: "https://github.com/dbuenzli/cmdliner/issues"
 tags: [ "cli" "system" "declarative" "org:erratique" ]
 license: "BSD3"
-available: [ocaml-version >= "4.00.0"]
+available: [ocaml-version >= "3.12.1"]
+patches: ["backport_pre_4_00_0.patch" {ocaml-version < "4.00.0"]
```
Comment (samoht) [inline packages/cmdliner/cmdliner.0.9.7/opam]: missing closing `}`

**Comment by dbuenzli** [comment]
Ah thanks Thomas, odd that `1.2` didn't complain. Looks better now.

**Pull request merged**